tests/lib/cpp/cxx/testcase.yaml

```diff
@@ -73,3 +73,11 @@ tests:
     build_only: true
     extra_configs:
       - CONFIG_STD_CPP2B=y
+  # Verify struct _cpu_arch struct _thread_arch are compatible with C++ when
+  # specific configs are enabled.
+  cpp.main.empty_struct_size:
```

Comment:
The test is arch agnostic, but the configs required to trigger the empty struct are architecture dependent. We could put an `arch_allow: riscv` on the test to make that clear. What do you think?